### PR TITLE
Make the 'time to update' message depend up on the channel.

### LINKED ISSUE #24158: Version staleness check too aggressive for stable channel
https://github.com/flutter/flutter/blob/ae2b215efe257485ba790301ff81aaf426a8445c/packages/flutter_tools/lib/src/version.dart#L217

Once we have a `stable` channel, people on that channel will start seeing a "WARNING: your installation of Flutter is X days old." message after 35 days, but there'll be no newer version available since that channel will only receive updates approximately once a quarter.

Ideally, we'd tailor that age check to the channel.  A dumb (but better than nothing) alternative would be just to bump that age check up to 100 days...

### PR DESCRIPTION
**Pull request by gspencergoog**
Fixes #24158

### CODE REVIEW

**Review by tvolkert — APPROVED** [review]

File: packages/flutter_tools/lib/src/version.dart [hunk 1]
```diff
@@ -207,20 +207,34 @@ class FlutterVersion {
...
   ///
-  /// This is set to 5 weeks because releases are currently around every 4 weeks.
+  /// For example, for the beta channel, this is set to five weeks because
+  /// beta releases happen approximately every month.
   @visibleForTesting
-  static const Duration kVersionAgeConsideredUpToDate = Duration(days: 35);
+  static Duration versionAgeConsideredUpToDate(String channel) {
+    switch (channel) {
+      case 'stable':
+        return const Duration(days: 7 * 4 * 4);
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/version.dart]: i recommend making this at least 6 months, maybe 12

File: packages/flutter_tools/lib/src/version.dart [hunk 2]
```diff
@@ -207,20 +207,34 @@ class FlutterVersion {
...
+      case 'beta':
+        return const Duration(days: 7 * 5);
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/version.dart]: lets make this more like 8 weeks

File: packages/flutter_tools/lib/src/version.dart (hunk 1 above)
Comment (gspencergoog) [reply packages/flutter_tools/lib/src/version.dart]: Done

File: packages/flutter_tools/lib/src/version.dart (hunk 2 above)
Comment (gspencergoog) [reply packages/flutter_tools/lib/src/version.dart]: Done.

**Pull request merged**